### ISSUE TITLE
FSE: Copying a page does not show header and footer

### ISSUE THREAD

**Issue by arunsathiya** [Automattic/wp-calypso#35605]
#### Steps to reproduce 

- Start with a FSE site
- Visit https://wordpress.com/pages
- Use the `Copy` option under the menu available next to a page
- A page template selector appears. Close this selector using the "X" at the top right. Related https://github.com/Automattic/wp-calypso/issues/35604 where it's suggested that the page templates should not be shown.

#### What is expected 

The new page looks the same as the old page with header and footer.

#### What happens instead 

The header and footer sections are missing. Only `Content` section's blocks are copied over.

<img width="1848" alt="Screenshot 2019-08-20 at 16 23 27" src="https://user-images.githubusercontent.com/18581859/63341552-17cefe00-c367-11e9-9a69-9885a6d9ad11.png">

**Comment by noahtallen** [comment]
Can reproduce. After refreshing the page, the header/footer display. Probably not merging template/post when copying

**Comment by noahtallen** [comment]
My hypothesis is that our filter doesn't get executed at the right time. My guess is that the flow looks like this:

1. User clicks on the "copy" button, which directs us to the new post page with the 'jetpack-copy' param set.
2. The new post is created.
3. Our template parts are inserted into the post object.
4. The jetpack_copy function gets called which updates the new post with the post content from the old one.
5. The user sees the page with the copied post content data, as the `the-post` action was not fired again.

**Comment by noahtallen** [comment]
I think this actually comes down to the following code:

https://github.com/WordPress/gutenberg/blob/b4c7054da14d2b4660a96d95998702fe599f316b/packages/editor/src/store/actions.js#L159-L189

### Some facts:
* When copying a post, `edits.content` will equal the post content of the copied post without the header/footer/post content blocks from FSE.
* However, also at this point, `post.content.raw` will contain the header/footer/post content block including the new copied data. This will be ignored in favor of the content from initial edits above.
* This is a new post (status is `auto-draft`).
* If a template exists and we have a new post, we try to reconcile the template with the initial edits (or the raw post content if there are no initial edits)

### How Jetpack Handled This
In https://github.com/Automattic/jetpack/pull/11295, @kwight added a bit of code to `copy_post` which removes the initial post template. As a result, since there is no template, Gutenberg is able to initialize the editor with the initial edits from copy_post.

### Why this is problematic for us
Initial edits does not contain the FSE site blocks. That's why we get this bug -- the editor is initialized with the copied initial edits **before** `the_post` action is called to merge our FSE blocks in. This data does not come from the API, it comes from the editor initialization code here: https://github.com/WordPress/WordPress/blob/1d6bd71ce4eda6fd4842754b222a56d45fc3ffa2/wp-admin/edit-form-blocks.php#L81-L109

**SPECIFICALLY**, the code above _ignores_ any API prefetch operations which modify the global post object. Even if FSE code fired at some point there, it would be ignored.

One solution I tried was to increase the priority of our `block_editor_settings` filter so that when we add the FSE template, Gutenberg will get it. This gives us the opposite bug. Instead of only the post content displaying, now only the template parts display. Why? Since it is both a new post and we now set our template correctly, it attempts to `synchronizeBlocksWithTemplate`. The template that we set is telling the editor that we want a template part, post content, followed by another template part. It tries to validate that against initialEdits, and discards all of the edits in favor of the template part. (Note, even if we make initialEdits have the FSE post content, we may run into the following bug: https://github.com/WordPress/gutenberg/issues/11681)

### TLDR; How to fix this?
We need to make sure that initialEdits includes the FSE blocks.

**Comment by noahtallen** [comment]
I was able to put together some filters to ensure that default content includes the blocks. However, like I thought, we still run into issues with block validation. Namely:

https://github.com/WordPress/gutenberg/blob/e7d674dc5ee245a41a8c2e9fac88a021dbaded32/packages/blocks/src/api/templates.js#L53-L55

We need to change this line to `( ! template || ! template.length )`

**Comment by davipontesblog** [comment]
Closing in favor of the core WordPress FSE implementation in dotCom.
